api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -507,15 +507,14 @@ private User initializeUserIfNeeded() {
     try {
       Integer moodleId = user.getMoodleId();
       if (moodleId == null) {
-        try {
-          moodleId = complianceService.getMoodleId(user.getEmail());
-          user.setMoodleId(moodleId);
-        } catch (ApiException ex) {
-          log.severe(String.format(" Error while retrieving Moodle Id %s ", ex.getMessage()));
-          throw new ServerErrorException(ex);
+        moodleId = complianceService.getMoodleId(user.getEmail());
+        if (moodleId == null) {
+          return ResponseEntity.ok(profile);
```

Comment:
OK one last nitpick: I'd include a comment here clarifying why it's ok to return an OK response, i.e. "We often expect to hit this point, when a new AoU user hasn't yet visited Moodle, they won't have a user ID yet. In this case, we simply return early because no synchronization needs to happen."